gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -887,6 +887,59 @@ def test_{{ method_name }}_rest(request_type):
     {% if not field.oneof or field.proto3_optional %}
     {# ignore oneof fields that might conflict with sample_request #}
     request_init["{{ field.name }}"] = {{ field.merged_mock_value(method.http_options[0].sample_request(method).get(field.name)) }}
+    # The version of a generated dependency at test runtime may differ from the version used during generation
```

Comment:
Do we need this logic in the ads templates?

Reply:
Yes, good catch! I added the fix in [e689b13](https://github.com/googleapis/gapic-generator-python/pull/1749/commits/e689b136d48371a6cd736c10c44341b31b6aee96) and filed https://github.com/googleapis/gapic-generator-python/issues/1788 to add constraints files to the ads templates, like we have in the default templates.